snakemake/common.py

```diff
@@ -54,6 +55,43 @@ def get_uuid(name):
     return uuid.uuid5(UUID_NAMESPACE, name)
 
 
+def get_file_hash(filename, algorithm="sha256"):
+    """find the SHA256 hash string of a file. We use this so that the
+       user can choose to cache working directories in storage.
+    """
+    from snakemake.logging import logger
+
+    if os.path.exists(filename):
+
+        # The algorithm must be available
+        try:
+            hasher = hashlib.new(algorithm)
+        except ValueError as ex:
+            logger.error("%s is not an available algorithm." % algorithm)
+            raise ex
+
+        with open(filename, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hasher.update(chunk)
+        return hasher.hexdigest()
+
+    logger.warning("%s does not exist." % filename)
+    raise FileNotFoundError
+
+
+def bytesto(bytes, to, bsize=1024):
```

Comment:
I think this is not needed anymore?

Reply:
I still use it under setting the workflow sources:

```python
        def check_size(f):
            """A helper function to check the filesize, and return the file
               to the calling function
            """
            gb = bytesto(os.stat(f).st_size, "g")
            if gb > warning_size_gb:
                logger.warning(
                    "File {} (size {} GB) is greater than the {} GB suggested size "
                    "Consider uploading larger files to storage first.".format(
                        f, gb, warning_size_gb
                    )
                )
            return f

```

Reply:
And now this is a function provided by the workflow class.